server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxQueryMethodContract.scala

```diff
@@ -0,0 +1,411 @@
+           |    "methodResponses": [
+           |        [
+           |            "Mailbox/query",
+           |            {
+           |                "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |                "queryState": "${generateQueryState(mailboxId)}",
+           |                "canCalculateChanges": false,
+           |                "ids": [
+           |                    "${mailboxId.serialize}"
+           |                ],
+           |                "position": 0,
+           |                "limit": 256
+           |            },
+           |            "c1"
+           |        ]
+           |    ]
+           |}""".stripMargin)
+  }
+
+  @Test
+  def queryByRoleShouldNotReturnDelegatedMailboxes(server: GuiceJamesServer): Unit = {
+    val andreInbox = MailboxPath.inbox(ANDRE)
+    val andreInboxId: MailboxId = server.getProbe(classOf[MailboxProbeImpl])
+      .createMailbox(andreInbox)
+    val bobInboxId: MailboxId = server.getProbe(classOf[MailboxProbeImpl])
+      .createMailbox(MailboxPath.inbox(BOB))
+
+    server.getProbe(classOf[ACLProbeImpl])
+      .replaceRights(andreInbox, BOB.asString, MailboxACL.FULL_RIGHTS)
+
+    val request =
+      s"""{
+         |  "using": [
+         |    "urn:ietf:params:jmap:core",
+         |    "urn:ietf:params:jmap:mail",
+         |    "urn:apache:james:params:jmap:mail:shares"],
+         |  "methodCalls": [[
+         |    "Mailbox/query",
+         |    {
+         |      "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+         |      "filter": {"role":"Inbox"}
+         |    },
+         |    "c1"]]
+         |}""".stripMargin
+
+      val response = `given`
+        .header(ACCEPT.toString, ACCEPT_RFC8621_VERSION_HEADER)
+        .body(request)
+      .when
+        .post
+      .`then`
+        .statusCode(SC_OK)
+        .contentType(JSON)
+        .extract
+        .body
+        .asString
+
+      assertThatJson(response)
+        .isEqualTo(
+        s"""{
+           |    "sessionState": "75128aab4b1b",
+           |    "methodResponses": [
+           |        [
+           |            "Mailbox/query",
+           |            {
+           |                "accountId": "29883977c13473ae7cb7678ef767cbfbaffc8a44a6e463d971d23a65c1dc4af6",
+           |                "queryState": "${generateQueryState(bobInboxId)}",
+           |                "canCalculateChanges": false,
+           |                "ids": [
+           |                    "${bobInboxId.serialize}"
+           |                ],
+           |                "position": 0,
+           |                "limit": 256
+           |            },
+           |            "c1"
+           |        ]
+           |    ]
+           |}""".stripMargin)
+  }
+
+  @Test
+  def queryByRoleShouldNotReturnDelegatedMailboxesWhenCaseVariation(server: GuiceJamesServer): Unit = {
```

Comment:
Your test above already proves we don't return delegated mailboxes.

And the case variation here is on bob's inbox? So wouldn't it be better to drop the delegated part and only test the case variation?

Reply:
> Your test above already proves we don't return delegated mailboxes.

I was unsure about that behaviour, it could have gone wrong (if the fallback mechanism doing a mailbox search is implemented with the wrong parameters). I want to keep that test.

> And the case variation here is on bob's inbox?

+1 

>  So wouldn't it be better to drop the delegated part and only test the case variation?

That's two separated tests